singularity-opac/neutrinos/mean_opacity_neutrinos.hpp

```diff
@@ -68,68 +86,45 @@ class MeanOpacity {
+              kappaPlanckNum += weight * alpha / rho * B * nu * dlnu;
+              kappaPlanckDenom += weight * B * nu * dlnu;
+
+              // Only contributions to integral from non-zero kappa
+              if (alpha > singularity_opac::robust::SMALL()) {
+                kappaRosselandNum +=
+                    weight * singularity_opac::robust::ratio(rho, alpha) *
+                    dBdT * nu * dlnu;
+                kappaRosselandDenom += weight * dBdT * nu * dlnu;
+              }
             }
 
-            // Trapezoidal rule
-            const Real nu0 = fromLog_(lnuMin);
-            const Real nu1 = fromLog_(lnuMax);
-            kappaPlanckNum -=
-                0.5 * 1. / rho *
-                (opac.AbsorptionCoefficient(rho, T, Ye, type, nu0, lambda) *
-                     opac.ThermalDistributionOfTNu(T, type, nu0) * nu0 +
-                 opac.AbsorptionCoefficient(rho, T, Ye, type, nu1, lambda) *
-                     opac.ThermalDistributionOfTNu(T, type, nu1) * nu1) *
-                dlnu;
-            kappaPlanckDenom -=
-                0.5 *
-                (opac.ThermalDistributionOfTNu(T, type, nu0) * nu0 +
-                 opac.ThermalDistributionOfTNu(T, type, nu1) * nu1) *
-                dlnu;
-            kappaRosselandNum -=
-                0.5 * rho *
-                (singularity_opac::robust::ratio(
-                     1., opac.AbsorptionCoefficient(rho, T, Ye, type, nu0,
-                                                    lambda)) *
-                     opac.DThermalDistributionOfTNuDT(T, type, nu0) * nu0 +
-                 singularity_opac::robust::ratio(
-                     1., opac.AbsorptionCoefficient(rho, T, Ye, type, nu1,
-                                                    lambda)) *
-                     opac.DThermalDistributionOfTNuDT(T, type, nu1) * nu1) *
-                dlnu;
-            kappaRosselandDenom -=
-                0.5 *
-                (opac.DThermalDistributionOfTNuDT(T, type, nu0) * nu0 +
-                 opac.DThermalDistributionOfTNuDT(T, type, nu1) * nu1) *
-                dlnu;
-
-            Real lkappaPlanck = toLog_(singularity_opac::robust::ratio(
-                kappaPlanckNum, kappaPlanckDenom));
-            Real lkappaRosseland = toLog_(singularity_opac::robust::ratio(
-                kappaRosselandDenom, kappaRosselandNum));
-            lkappaPlanck_(iRho, iT, iYe, idx) = lkappaPlanck;
-            lkappaRosseland_(iRho, iT, iYe, idx) = lkappaRosseland;
+            Real kappaPlanck = singularity_opac::robust::ratio(
+                kappaPlanckNum, kappaPlanckDenom);
+            Real kappaRosseland =
+                kappaPlanck > singularity_opac::robust::SMALL()
+                    ? singularity_opac::robust::ratio(kappaRosselandDenom,
+                                                      kappaRosselandNum)
+                    : 0.;
+            printf("kappas: %e %e\n", kappaPlanck, kappaRosseland);
```

Comment:
Should this printf still be here? If so might be good to make it more descriptive

Reply:
Oops yeah that was garbage, gone